File: bms/go/geo/geo_r2_Rect_AddPoint/geo_r2_Rect_AddPoint.go
package test

var ()

const ()

func (r geo_r2_Rect) AddPoint(p geo_r2_Point) geo_r2_Rect {
	return geo_r2_Rect{r.X.AddPoint(p.X), r.Y.AddPoint(p.Y)}
}

func (i geo_r1_Interval) AddPoint(p float64) geo_r1_Interval {
	if i.IsEmpty() {
		return geo_r1_Interval{p, p}
	}
	if p < i.Lo {
		return geo_r1_Interval{p, i.Hi}
	}
	if p > i.Hi {
		return geo_r1_Interval{i.Lo, p}
	}
	return i
}

func (i geo_r1_Interval) IsEmpty() bool { return i.Lo > i.Hi }

type geo_r2_Rect struct {
	X, Y geo_r1_Interval
}

type geo_r1_Interval struct {
	Lo, Hi float64
}

type geo_r2_Point struct {
	X, Y float64
}

File: translations_only/success/go/geo/geo_r2_Rect_AddPoint/claude2/CAPR/geo_r2_Rect_AddPoint.rs

use std::ops::Add;

struct GeoR2Point {
    x: f64,
    y: f64,
}

struct GeoR1Interval {
    lo: f64,
    hi: f64,
}

struct GeoR2Rect {
    x: Box<GeoR1Interval>,
    y: Box<GeoR1Interval>, 
}

fn add_point(r: GeoR2Rect, p: GeoR2Point) -> GeoR2Rect {
    GeoR2Rect {
        x: Box::new(add_point_interval(*r.x, p.x)),
        y: Box::new(add_point_interval(*r.y, p.y)),
    }
}

fn add_point_interval(mut i: GeoR1Interval, p: f64) -> GeoR1Interval {
    if i.lo > i.hi {
        return GeoR1Interval { lo: p, hi: p };
    }
    if p < i.lo {
        return GeoR1Interval { lo: p, hi: i.hi };
    }
    if p > i.hi {
        return GeoR1Interval { lo: i.lo, hi: p };
    }
    i
}

fn is_empty(i: &GeoR1Interval) -> bool {
    i.lo > i.hi
}

